ochre/utils/hpxml.py

```diff
@@ -371,34 +371,46 @@ def parse_hpxml_boundaries(hpxml, return_boundary_dicts=False, **kwargs):
     else:
         raise IOError(f'Unable to parse multiple floor areas: {main_floor_areas}')
 
+    attic_floor_area = first_floor_area  # may get updated later
     if 'Garage Floor' in boundaries:
         garage_floor_area = boundaries['Garage Floor']['Area (m^2)'][0]
         attached_wall_lengths = [area / ceiling_height for area in boundaries['Garage Attached Wall']['Area (m^2)']]
-        if len(attached_wall_lengths) == 1:
+        attached_wall_azimuths = [wall['Azimuth'] % 180 for wall in attached_walls.values()]
+        n_sides = len(set(attached_wall_azimuths))
+        if n_sides == 1:
             garage_area_in_main = 0
-        elif len(attached_wall_lengths) == 2:
-            # at least part of garage is included in the main house footprint (note, house may not have 2nd floor)
-            garage_area_in_main = attached_wall_lengths[0] * attached_wall_lengths[1]
-            assert 0.1 < garage_area_in_main / garage_floor_area < 0.9  # should be close to 50% for ResStock cases
+            if total_floors == 1:
+                attic_floor_area += garage_floor_area
+        elif n_sides == 2:
+            # at least part of garage is included in the main house footprint
+            if len(attached_wall_lengths) == 2:
+                # usually for 1-story home or garage that is fully under the 2nd story
+                garage_area_in_main = attached_wall_lengths[0] * attached_wall_lengths[1]
```

Comment:
We should add a TODO to check this always works. Garage with a flat roof could protrude and break this.

Reply:
Not sure a flat roof would break it, but we should definitely create a test suite to check

Reply:
Just added a test case to Teams